### ISSUE TITLE
Leader Support

### ISSUE THREAD

**Issue by jimmiebfulton** [zellij-org/zellij#2890]
While the modal approach, a la Vim, is very nice, I think we may have thrown the baby out with the bath water.  Vim does't just have the modal editing, but it has a leader system.  Tmux, if I'm not mistaken, isn't modal, but is definitely leader driven.   The best of all worlds would be to have our cake, and eat it too.  Full blown modal with leader support.

The locking feature is basically a reverse leader.  But with this approach, you have a couple of options:

1) Use lock mode as a poor-man's leader system, leaving your interface locked most of the time.  You unlock, switch into a mode and do your thing, and then lock again.  It's an extra step that begs for a counter-mechanism... a leader.
2) Switching to locked mode is too inconvenient.  Leave it out of lock mode, and remap all the keys in either Vim or in Zellij so they don't collide.  Then, lock mode is pretty much useless except for extreme cases.

These approaches are pretty much mutually exclusive.

Here's what I, and I'm guessing many others would like... I'm going to put in hypothetical binding, but these obviously should be configurable to whatever taste.  I'm also including aspects of my own tmux config; this is actually how I work in TMux, minus the modal parts of Zellij:

Set up my leader as "<C-Space>".  Zellij ONLY responds to this leader, and no other key combinations.

"<leader>t": switch into Tab modal.  Do my tab stuff and escape.  I've effectively locked Zellij. 
"<leader>p": switch into Pane modal...  You get the idea.
"<leader><leader>": Paint numbers over the Panes I can switch to, and hitting a number switches to that pane and relocks Zellij.

In this way, I'd use "<C-Space>" to enter into Zellij, and otherwise it doesn't conflict with my other bindings.  "<Space>" enters into Neovim space.  Two systems that operate the same, with different but similar binding.

I haven't looked at the code base yet, but I'm guessing all of the ingredients are there.  I may have some interest in assisting with this, as well, provided there is interest by the team in adding this kind of capability, and doesn't collide with an established vision.

**Comment by vihu** [comment]
Concur, I would also much prefer this approach. But rather than calling it a "leader" I'd just call it "prefix", basically exactly the same thing as tmux. The best thing about this approach is that there won't be any conflicts with terminal/vim/nvim ever (that's pretty much the primary thing keeping me from switching to zellij full time from tmux).

It could be an additional/alternate way to map keybinds along with the existing one.

So we'd do `prefix + key` for executing keybinds. For example `ctrl + b, |` to split vertically, `ctrl + b, -` to split horizontally and so on. This would keep the mental overhead of learning new keybinds to a minimum as well (i.e. just use the same keybinds as your existing tmux and rely on the already built up decade+ old muscle memory :) ).

**Comment by musjj** [comment]
Would love this too.

`Ctrl p` is a very critical keybind in bash and basically all applications based on readline, so it's really annoying that it's used by default. `Ctrl t` is also used by fzf's default shell integration. I tried to explore some alternative keybinds, but I could find a good one that isn't used by something else.

**Comment by jimmiebfulton** [comment]
I’ve switched to WezTerm. Super flexible scripting model (lua) allows me to replace Alacritty and tux in one very nice package.

**Comment by musjj** [comment]
I was initially attracted by zellij's layouting system. I'm still not to deep into terminal multiplexing in general, but so far I've found it superior compared to the one in kitty and wezterm. Haven't tried tmux yet though.

**Comment by musjj** [comment]
I figured out a way to simulate leader keys with the current keybinding system. The trick is to have `Locked` as the default mode and use your "leader" key as a switch to normal mode. Then, adjust the rest of the keybindings accordingly.

Here's my config if anyone want to try: https://gist.github.com/musjj/1f816620a2b46c95a0b243569c457af0

The leader key is `Ctrl g` and the top level keys looks like this:
```
bind "p" {
  SwitchToMode "Pane"
}
bind "m" {
  SwitchToMode "Move"
}
bind "t" {
  SwitchToMode "Tab"
}
bind "r" {
  SwitchToMode "Resize"
}
bind "s" {
  SwitchToMode "Scroll"
}
bind "o" {
  SwitchToMode "Session"
}
```
The rest remains like the default.

**Comment by arexon** [comment]
Perhaps I'm missing something, but wouldn't the `Tmux` mode bindings and the `clear-defaults` option solve this?

```kdl
keybinds {
    Normal clear-defaults=true {
        bind "Alt a" { SwitchToMode "Tmux"; }
    }
    Tmux clear-defaults=true {
        bind "v" { NewPane "Right"; SwitchToMode "Normal"; }
        bind "s" { NewPane "Down"; SwitchToMode "Normal"; }
        // Define the rest of the keybinds here..
    }
    // Clear the other modes' keybinds..
    Pane clear-defaults=true {}
    Tab clear-defaults=true {}
    Resize clear-defaults=true {}
}
```

Granted, it's not an ideal solution but it fixed all of my problems with Zellij's keybinds.

**Comment by musjj** [comment]
Yeah, that's what I did too, except with `Locked` instead of `Tmux`. I guess you can basically use any modes you want for this. One downside I'm seeing is that the status bar isn't displaying the keybinds correctly for some reason, but I don't really use it so w/e.
